src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Rfc2898DeriveBytes.OneShot.cs

```diff
@@ -329,10 +329,8 @@ public partial class Rfc2898DeriveBytes
 
         private static void ValidateHashAlgorithm(HashAlgorithmName hashAlgorithm)
         {
-            if (string.IsNullOrEmpty(hashAlgorithm.Name))
-                throw new ArgumentException(SR.Cryptography_HashAlgorithmNameNullOrEmpty, nameof(hashAlgorithm));
-
-            string hashAlgorithmName = hashAlgorithm.Name;
+            string? hashAlgorithmName = hashAlgorithm.Name;
+            ArgumentException.ThrowIfNullOrEmpty(hashAlgorithmName, nameof(hashAlgorithm));
```

Comment:
is compiler smart enough to recognize that `hashAlgorithmName` is not `null` after the call to `ArgumentException.ThrowIfNullOrEmpty`? I find using nullable variable a little bit confusing here (but that is just my personal opion).

```suggestion
            ArgumentException.ThrowIfNullOrEmpty(hashAlgorithm.Name, nameof(hashAlgorithm));
            string hashAlgorithmName = hashAlgorithm.Name;
```

Reply:
> is compiler smart enough to recognize that hashAlgorithmName is not null after the call to ArgumentException.ThrowIfNullOrEmpty?

Yes, because the argument to ThrowIfNullOrEmpty is annotated as `[NotNull] string?`, which means if the method successfully returns, the value passed in wasn't null.